python/amici/ode_export.py

```diff
@@ -1636,6 +1696,9 @@ def _equation_from_component(self, name: str, component: str) -> None:
         self._eqs[name] = sp.Matrix(
             [comp.get_val() for comp in getattr(self, component)]
         )
+
+        if name == 'x0':
```

Comment:
I'm not sure I understand the full logic yet how this is implemented, but this shouldn't be handled here. After initialization, the ODEExporter should be completely independent of the SbmlImporter, since it can also handle other model types (currently only PYSB models).